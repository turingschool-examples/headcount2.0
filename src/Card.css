
.card {
  width: 150px;
  text-align: center;
  margin: 2%;
  background: white;
  opacity: .86;
  border-radius: 5px;
  padding: 10px;
  box-shadow: 3px 3px 3px #7c7c7c;
}

h1 {
  font-size: 20px;
}

.data {
  font-size: 15px;
  text-align: left;
  font-weight: bold;
}

.less-than {
  color: #ea6512;
}

.greater-than {
  color: #4f4f4f;
}
