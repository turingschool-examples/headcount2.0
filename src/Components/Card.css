.card {
  background-color: #375053;
  border-radius: 15px;
  box-shadow: 0px 0px 19px 1px rgba(0, 0, 0, .7);
  flex: 1 18em;
  list-style-type: none;
  margin: 0 2em 2em 2em;
  padding: 0 1em 1em 1em;
}

.selected {
  background-color: #5e7377;
  border: solid 4px #f7b488;
}

h2 {
  font-size: 2em;
  margin: .5em;
  text-align: center;
}

li {
  font-size: 1.2em;
  padding: 0 0 0 6em;
}

.stat {
  padding: 0 0 .5em 0;
}

.less-than {
  color: #f7b488;
}