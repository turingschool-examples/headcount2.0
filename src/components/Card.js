import React from 'react';
import PropTypes from 'prop-types';


const Card = ({ location, data, handleCompare, compare, active }) => {
  const classname = active === true ? 'card active' : 'card'
  return (
    <div className={classname} key={location} onClick={() => handleCompare(location)}>
      <h1 className='cardTitle'>
        {location}
      </h1>
      {Object.keys(data).map((key, index) => {


        return (
          <div className='cardData' key={index}>
            <li className={
              data[key] <= 0.5 ? 'negative' : 'positive'
            }> {key}: {data[key]} </li>
          </div>
        )
      })}
    </div>
  );
}

Card.propTypes = {
  location: PropTypes.string,
  data: PropTypes.object,
  handleCompare: PropTypes.func,
}

export default Card;
