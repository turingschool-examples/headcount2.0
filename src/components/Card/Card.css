/*CARD COMPONENT RELATED CSS GOES HERE*/
.Card {
  height: 350px;
  width: 250px;
  margin: 5px 0;
  padding: 0.5rem;
  padding-bottom: 1.25rem;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 4rem 1fr;
  justify-items: center;
  color: #006989;
  letter-spacing: 0.5px;
  border-radius: 5%;
  background: rgb(255, 255, 255);
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
  box-sizing: border-box;
}

.Card h2 {
  text-align: center;
  grid-row: 1 / 2;
  font-family: "Roboto Condensed", sans-serif;
  font-weight: 700;
  font-size: 20px;
  margin: 0;
  padding-top: 1.25rem;
}

ul {
  grid-row: 2 / 3;
  list-style-type: none;
  font-family: "Roboto", sans-serif;
  font-weight: 400;
  text-size: 1.5rem;
  padding-left: 0;
  align-self: center;
}

.low {
  color: #EF713B;
}

.selected {
  border: 5px solid #EF713B;
}