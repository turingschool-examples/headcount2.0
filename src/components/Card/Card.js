import React from 'react';
import PropTypes from 'prop-types';

const Card = props => {
  const { location, stats } = props;

  const districtStats = Object.keys(stats).map(stat => <li key={stat}>{stats[stat]}</li>);

  return (
    <div>
      <h2>{location}</h2>
      <ul>
        {districtStats}
      </ul>
    </div>
  );
};

Card.propTypes = {
  location: PropTypes.string.isRequired,
  stats: PropTypes.object.isRequired
};

export default Card;
