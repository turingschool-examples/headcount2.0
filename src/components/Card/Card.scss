$backgroundColor: rgba(255, 255, 255, .1);
$sublimeGreen: #AAD74F;
$sublimeRed: #D23D54;
$sublimeBlue: #78C9D0;
$sublimeYellow: #FFFF74;

article {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: 20px;
  width: 250px;
  border: 1px solid #fff;
  background: $backgroundColor;
}

article h3 {
  color: $sublimeGreen;
  font-size: 1.5em;
  margin-bottom: 0px;
}

article p {
  margin: 7px;
}

ul {
  text-align: left;
  list-style: none;
  padding: 0;

  li {
    margin: 3px 3px;
  }

  .over {
    color: $sublimeBlue;
  }

  .under {
    color: $sublimeRed;
  }

  .fa {
    margin-right: 8px;
  } 

  .fa-bolt {
    color: yellow;
  }

  .fa-tint {
    color: #849FED;
  }
}

.clicked {
  border: 2px solid $sublimeYellow;
}