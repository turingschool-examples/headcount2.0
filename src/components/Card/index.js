import React, { Component } from 'react'
import PropTypes from 'prop-types'
import './Card.css'

class Card extends Component {
  constructor({location, stats, adjustComparisons}) {
    super()
    this.state = {
      location: location,
      stats: stats,
      isSelected: false
    }
  }

  makeCards = stats => {
    const isSelected = this.state.isSelected

    return Object.keys(stats).map((year, index) => {
      return (
        <li key={index}
            className={`Card-entry ${this.checkLow(stats, year)}`}
        >
          {year}: {stats[year]}
        </li>)
      }
    )
  }

  checkLow = (stats, year) => {
    if (stats[year] <= 0.5) return 'low'
  }

  checkSelected = isSelected => {
    if (isSelected) {
      return 'selected'
    } else {
      return ''
    }
  }

  toggleSelect = event => {
    const card = event.target.parentElement.find('.Card')

    debugger

    this.setState({ isSelected: !this.state.isSelected })
    this.props.adjustComparisons(card)
  }

  render() {
    const location = this.state.location
    const stats = this.state.stats
    const isSelected = this.state.isSelected

    return(
      <article
        className={`Card ${this.checkSelected(isSelected)}`}
        onClick={this.toggleSelect}
      >
        <h2 className="Card-title">{location}</h2>
        <ul className="Card-list">
          { stats ? this.makeCards(stats) : null}
        </ul>
      </article>
    )
  }
}

Card.propTypes = {
  location: PropTypes.string.isRequired,
  stats: PropTypes.object
}

export default Card