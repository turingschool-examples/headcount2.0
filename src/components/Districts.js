import React from 'react';
import Card from './Card';

const Districts = ({ stats, setSelectedCard, selectedCard }) => {

  const statKeys = Object.keys(stats);
  const districtCards = statKeys.map((stat, index) => {
    return <Card {...stats[stat]} key={index + 'district'} setSelectedCard={setSelectedCard} />
  });

  return (
    <section className="districtCards">
        {districtCards}
    </section>
  );
};

export default Districts;