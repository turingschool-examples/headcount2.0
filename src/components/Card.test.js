import React from 'react';
import Districts from './Districts';
import Card from './Card';
import {shallow, mount} from 'enzyme';

describe('Card', () => {
  
  it('should match the snapshot', () => {
    const mockStats ={
      'COLORADO' : {
        location: 'COLORADO',
        stats: {},
        selected: false
      }};

    const wrapper = shallow(<Card stats={mockStats} setSelectedCard={jest.fn()} selectedCards={[]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('adds class of selected to Card', () => {
    const mockLocation ='COLORADO';
    const mockStats = {};
    const selected = true;
    const setSelectedCard = jest.fn();
  
    const wrapper = shallow(<Card location={mockLocation} stats={mockStats} setSelectedCard={setSelectedCard} selected={selected} />);
    
    expect(wrapper.find(".selected").length).toEqual(1);
  });

  it('calls selectedState on Click', () => {
    const mockLocation ='COLORADO';
    const mockStats = {};
    const selected = true;
    const setSelectedCard = jest.fn();
    const wrapper = mount(<Card location={mockLocation} stats={mockStats} setSelectedCard={setSelectedCard} selected={selected} />);
    const spy = spyOn(wrapper.instance(), setSelectedCard)
    
    wrapper.find('article').simulate('click');
    expect()
    
  });
}); 