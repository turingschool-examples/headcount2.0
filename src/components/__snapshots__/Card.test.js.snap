// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    selectedCards={Array []}
    setSelectedCard={[Function]}
    stats={
        Object {
            "COLORADO": Object {
              "location": "COLORADO",
              "selected": false,
              "stats": Object {},
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2 />,
        <hr />,
        <ul>
          <li
                    className=""
          >
                    COLORADO
                    : 
                    <span>
                              Object {
                                        "location": "COLORADO",
                                        "selected": false,
                                        "stats": Object {},
                                      }
                    </span>
          </li>
</ul>,
      ],
      "className": "card ",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className=""
>
              COLORADO
              : 
              <span>
                            Object {
                                          "location": "COLORADO",
                                          "selected": false,
                                          "stats": Object {},
                                        }
              </span>
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "COLORADO",
                ": ",
                <span>
                  Object {
                                    "location": "COLORADO",
                                    "selected": false,
                                    "stats": Object {},
                                  }
</span>,
              ],
              "className": "",
            },
            "ref": null,
            "rendered": Array [
              "COLORADO",
              ": ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Object {
                    "location": "COLORADO",
                    "selected": false,
                    "stats": Object {},
                  },
                },
                "ref": null,
                "rendered": Object {
                  "location": "COLORADO",
                  "selected": false,
                  "stats": Object {},
                },
                "type": "span",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2 />,
          <hr />,
          <ul>
            <li
                        className=""
            >
                        COLORADO
                        : 
                        <span>
                                    Object {
                                                "location": "COLORADO",
                                                "selected": false,
                                                "stats": Object {},
                                              }
                        </span>
            </li>
</ul>,
        ],
        "className": "card ",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className=""
>
                COLORADO
                : 
                <span>
                                Object {
                                                "location": "COLORADO",
                                                "selected": false,
                                                "stats": Object {},
                                              }
                </span>
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "COLORADO",
                  ": ",
                  <span>
                    Object {
                                        "location": "COLORADO",
                                        "selected": false,
                                        "stats": Object {},
                                      }
</span>,
                ],
                "className": "",
              },
              "ref": null,
              "rendered": Array [
                "COLORADO",
                ": ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Object {
                      "location": "COLORADO",
                      "selected": false,
                      "stats": Object {},
                    },
                  },
                  "ref": null,
                  "rendered": Object {
                    "location": "COLORADO",
                    "selected": false,
                    "stats": Object {},
                  },
                  "type": "span",
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
